cratedb_toolkit/testing/testcontainers/cratedb.py

```diff
@@ -101,3 +135,65 @@ def _connect(self):
         #       In `testcontainers-java`, there is the `HttpWaitStrategy`.
         # TODO: Provide a client instance.
         wait_for_logs(self, predicate="o.e.n.Node.*started", timeout=MAX_TRIES)
+
+
+class TestDrive:
+    """
+    Use different schemas for storing the subsystem database tables, and the
+    test/example data, so that they do not accidentally touch the default `doc`
+    schema.
+    """
+
+    EXT_SCHEMA = "testdrive-ext"
+    DATA_SCHEMA = "testdrive-data"
+
+    RESET_TABLES = [
+        f'"{EXT_SCHEMA}"."retention_policy"',
+        f'"{DATA_SCHEMA}"."raw_metrics"',
+        f'"{DATA_SCHEMA}"."sensor_readings"',
+        f'"{DATA_SCHEMA}"."testdrive"',
+        f'"{DATA_SCHEMA}"."foobar"',
+        f'"{DATA_SCHEMA}"."foobar_unique_single"',
+        f'"{DATA_SCHEMA}"."foobar_unique_composite"',
+        # cratedb_toolkit.io.{influxdb,mongodb}
+        '"testdrive"."demo"',
+    ]
+
+
+class CrateDBFixture:
+    """
+    A little helper wrapping Testcontainer's `CrateDBContainer` and
+    CrateDB Toolkit's `DatabaseAdapter`, agnostic of the test framework.
+    """
```

Comment:
I appreciate very much bringing this into the generic `cratedb_toolkit.testing` module namespace, to make it a re-usable component for other packages.

Reply:
Talking about "naming things", if you have a better idea for the name, in order to better disambiguate from pytest's notion of "fixtures", because it is actually a testframework-agnostic adapter/wrapper around, well, DatabaseAdapter [^1] and CrateDBContainer.

[^1]: Also eventually to be renamed to `CrateDBClientAdapter` or something different.

Reply:
Would `CrateDBTestAdapter` be a more sensible name?

Reply:
Given that we have `DatabaseAdapter`, I like the `CrateDBTestAdapter` name!

Reply:
Wonderful. Let us rename it on behalf of a subsequent patch.

Reply:
I see you already renamed it properly, thanks.

I proposed to change it later, because I feared this would add too much noise to this patch because the symbol would need to be changed at too many places. But, of course, it turns out that it already has been nicely decoupled from the test cases on behalf of the `cratedb` and `cratedb_service` pytest fixtures, so my fears were unfounded.